### ISSUE TITLE
Loop not terminated by EOF

### ISSUE THREAD

**Issue by ludekstepan** [deoxxa/dissolve#7]
I have following structure:

```
Header
- frame 1
- frame 2
- frame ...N
[EOF]
```

The header does not contain number of frames, there's just EOF after the last frame. The problem is that the loop never finishes and I never get any data from parser. Any idea?

**Comment by articice** [comment]
Are you talking about parsing a file that ends at the end of last frame?
Or, are you talking about a special EOF/EOM character in the end?

**Comment by ludekstepan** [comment]
Yes, I mean a situation when a file ends exactly after a repeated frame.
